src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/HttpApi/ApiClient.cs

```diff
@@ -408,5 +427,25 @@ private static void AppendDuration(StringBuilder builder, TimeSpan duration)
                 builder.Append(Convert.ToInt32(duration.TotalSeconds).ToString(CultureInfo.InvariantCulture));
             }
         }
+
+        private string GetRequestPrefix(ProcessIdentifierFormat processIdentifierFormat)
```

Comment:
I like this idea, but let's take it a few steps further. Instead of having each method pass in the identifier (which they were already doing) and this extra ProcessIdentifierFormat parameter, how about we do something like:

`private static string GetProcessQuery(int? pid = null, Guid? uid = null, string name = null)`

and this will return "pid={pid}", "uid={uid}", "name={name}", or throw an exception if none are specified.

I'll use GetProcessAsync as an example of how to use this. Change GetProcessAsync to be:

```diff
public Task<Models.ProcessInfo> GetProcessAsync(int pid, CancellationToken token)
{
-   return GetProcessAsync(pid.ToString(CultureInfo.InvariantCulture), token);
+   return GetProcessAsync(GetProcessQuery(pid: pid), token);
}

public Task<Models.ProcessInfo> GetProcessAsync(Guid uid, CancellationToken token)
{
-   return GetProcessAsync(uid.ToString("D"), token);
+   return GetProcessAsync(GetProcessQuery(uid: uid), token);
}

-private async Task<Models.ProcessInfo> GetProcessAsync(string processKey, CancellationToken token)
+private async Task<Models.ProcessInfo> GetProcessAsync(string processQuery, CancellationToken token)
```

This gets each of the public methods out of the business of knowing how to format each of the process identifier parameters and centralizes the query parameter construction (even if it is fairly simple).